app/src/main/java/com/amaze/filemanager/filesystem/Operations.java

```diff
@@ -441,6 +441,61 @@ public static void rename(
 
       private final DataUtils dataUtils = DataUtils.getInstance();
 
+      private final Boolean isCaseSensitiveRename =
+          oldFile.getSimpleName().equalsIgnoreCase(newFile.getSimpleName())
+              && !oldFile.getSimpleName().equals(newFile.getSimpleName());
+
+      // random string that is appended to file to prevent name collision, max file name is 255
+      // bytes
+      private static final String TEMP_FILE_EXT = "u0CtHRqWUnvxIaeBQ@nY2umVm9MDyR1P";
+
+      private boolean localRename(@NonNull HybridFile oldFile, @NonNull HybridFile newFile) {
+        File file = new File(oldFile.getPath());
+        File file1 = new File(newFile.getPath());
+        boolean result = false;
+
+        switch (oldFile.getMode()) {
+          case FILE:
+            int mode = checkFolder(file.getParentFile(), context);
+            if (mode == 2) {
+              errorCallBack.launchSAF(oldFile, newFile);
```

Comment:
The errorCallback.launchSAF needs to be lifted out of this function or else another errorCallback will be called.